### PR TITLE
[WIP] Add Hawkular provider support

### PR DESCRIPTION
**Pull request by vnugent**
@psav First try at extending the framework to support Middleware/Hawkular page.  I was able to navigate to Add new Middleware provider page.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit ff3e7fae6a0be369df9b619a3751086d2af6d5dd:

`cfme/middleware/__init__.py`:
- :heavy_exclamation_mark: Line 4:46: **W292** *no newline at end of file*

`cfme/tests/middleware/test_provider_crud.py`:
- :heavy_exclamation_mark: Line 8:58: **W292** *no newline at end of file*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 648503117561d201f359d7e75740bdad23166c2d:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/tests/middleware/test_provider_crud.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from cfme.middleware.hawkular import HawkularProvider
+from cfme import login
+
+login.login_admin()
+
+hawkular = HawkularProvider('My Hawkular', 'livingontheedge.hawkular.org', '80')
+
+hawkular.create(cancel=False, validate_credentials=False)
```
Comment (mfalesni) [inline cfme/tests/middleware/test_provider_crud.py]: All these things must be enclosed in test functions :)

```python
import pytest

@pytest.fixture(scope='function')
def hawkular_provider():
    return HawkularProvider(...)

def test_hawkular_crud(hawkular_provider):
    hawkular_provider.create(cancel=False, validate_credentials=False)
    # ...
```

Ideally, the data for the HawkularProvider should live in cfme_data.yaml and the test just call something like `HawkularProvider.from_config("config_name")`

**Comment by vnugent** [comment]
@mattmahoneyrh @hhovsepy

File: cfme/tests/middleware/test_provider_crud.py (hunk 1 above)
Comment (vnugent) [inline cfme/tests/middleware/test_provider_crud.py]: Noted. I should clarify test_provider_crud.py was a poor man unit test.  Will turn it to proper py.test

**Comment by mattmahoneyrh** [comment]
To fix the Hawkular Provider create assert failure, add the following to HawkularProvider constructor:
    self.string_name = 'Middleware'

**Comment by dajoRH** [comment]
Lint report for commit 833c37bae1ce8ba706edf500fdc942219c1ae0cc:

`utils/providers.py`:
- :bangbang: Line 509:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 514:9: **E265** *block comment should start with '# '*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by vnugent** [comment]
I made some improvement and rebase'd.  For some reason my commits introduce a host of unrelated commits.  Any suggestions?

**Comment by dajoRH** [comment]
Lint report for commit f8bab4551c70f24e86d943cc7476f3f6654912a2:

`utils/providers.py`:
- :bangbang: Line 509:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 514:9: **E265** *block comment should start with '# '*
- :bangbang: Line 516:1: **E302** *expected 2 blank lines, found 1*
- :red_circle: Line 516:1: **F811** *redefinition of unused 'clear_middleware_providers' from line 509*
- :bangbang: Line 521:9: **E265** *block comment should start with '# '*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Pull request closed without merging**